src/System.Net.Security/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -24,11 +25,19 @@ internal static partial class NegotiateStreamPal
     {
         internal static IIdentity GetIdentity(NTAuthentication context)
         {
-            Debug.Assert(!context.IsServer, "GetIdentity: Server is not supported");
-
             string name = context.Spn;
             string protocol = context.ProtocolName;
 
+            if (context.IsServer)
+            {
+                var safeContext = context.GetContext(out var status);
+                if (status.ErrorCode != SecurityStatusPalErrorCode.OK)
+                {
+                    throw new Exception(status.ErrorCode.ToString()); // TODO?
```

Comment:
Need to decide what kind of exception to throw here.

Reply:
Probably a Win32Exception using this: https://github.com/dotnet/corefx/blob/39cc91592c3d9c8fe7700a6323755cd7747c34ab/src/Common/src/System/Net/Security/NegotiateStreamPal.Windows.cs#L194

Reply:
InvalidHandle is the only other return value from GetContext, and I found another case here where it's used directly with Win32Exception: 
https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Net.Security/src/System/Net/NTAuthentication.cs#L25